CONFIG_OPTIONS.md

```diff
@@ -0,0 +1,575 @@
+
+Here `optional` means you don't have to manually provide it in the config.yaml file, and that it will be set to the default value if not provided.
+
+## Design Intention
+The intention of this configuration file is to provide an interface by which
+all deployments of DC/OS, regardless of provider, have a similar format, thus
+complementing the goal of dcos-launch to provide a single tool for launching
+across a variety of provider APIs.
+
+## Supported Deployments and Examples
+
+See [sample_configs](dcos_launch/sample_configs/) for a full list.
+
+- [Simple AWS Cloudformation](dcos_launch/sample_configs/aws-cf.yaml)
+- [Zen AWS Cloudformation](dcos_launch/sample_configs/aws-zen-cf.yaml)
+- [Onprem Install on AWS Bare Cluster](dcos_launch/sample_configs/aws-onprem.yaml)
+- [Azure Template Deployment](dcos_launch/sample_configs/azure.yaml)
+- [Onprem Installation on Google Cloud Platform](dcos_launch/sample_configs/gcp-onprem-with-helper.yaml)
+- [GCP Onprem with fault-domain helper](dcos_launch/sample_configs/gcp-onprem-with-fd-helper.yaml)
+
+* `onprem` can only be provisioned via `aws` and `gcp` platforms
+* DC/OS deployed from aws or azure provider do not technically need `ssh_user` or `ssh_private_key_filename`. However, without this additional data, the integration tests will not be trigger-able from dcos-launch. Thus, it is not recommended, but allowable, to omit these fields when not using the onprem provider
+
+## Universal params
+
+These options are for any config
+
+### `provider`
+
+string, required
+
+Which provider you will deploy a cluster to.
+
+* `aws`: Uses Amazon Web Services (AWS) CloudFormation console. Supports both zen and simple templates. (Can only be used with `platform: aws`. Requires: `template_url`, `template_parameters`
+* `azure`: Uses Azure Resource Manager deployment templates. Supports both ACS (Azure Container Service) and DC/OS templates. (Can only be used with `platform: azure`. Requires `template_url`, and `template_parameters`
+* `onprem`: Uses the DC/OS bash installer to orchestrate a deployment on arbitrary hosts of a bare cluster. Requires `num_masters`, `num_private_agents`, `num_public_agents`, `installer_url`, `instance_type`, `os_name`, and `dcos_config`
+
+
+Allowed: aws, azure, acs-engine, onprem, terraform.
+
+### `launch_config_version`
+
+integer, required
+
+This is still a tool under active development and as such a strict version specifier must be included. (Right now there is only v1.)
+
+Allowed: 1
+
+### `ssh_port`
+
+integer, optional
+
+What port to use on SSH.
+
+Default: 22
+
+
+### `ssh_private_key_filename`
+
+string, optional (required if you are doing an onprem deploy)
+
+This is required if you don't use `key_helper` or `ssh_private_key`.
+
+If `key_helper: true` then this field cannot be supplied.
+
+### `ssh_private_key`
+
+string, optional
+
+If this is not set and you are not using `key_helper`, you must provide `ssh_private_key_filename`.
```

Comment:
Any interactions with `key_helper: true`?

Reply:
if `key_helper` is true then it will skip loading the `ssh_private_key` (using `ssh_private_key_filename`)

Reply:
It doesn't raise a ValueError though it probably should since if you're somehow trying to use key_helper and supply your own ssh key, that doesn't make sense